src/devices/GoPiGo3/Models/GrooveSensors.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.GoPiGo3.Models
+{
+    public class GrooveSensor
+    {
+        /// <summary>
+        /// Constructor for the groove sensor part of the main GoPiGo3 class
+        /// </summary>
+        /// <param name="groovePort">The Groove port, either Groove1 or Groove2</param>
+        public GrooveSensor(GroovePort groovePort)
+        {
+            if (!((groovePort != GroovePort.Groove1) || (groovePort != GroovePort.Groove2)))
+                throw new ArgumentException("Groove sensor can only be on Port 1 or Port 2");
+            SensorType = GrooveSensorType.None;
+        }
+
+        public GroovePort Port { get; }
+        public GrooveSensorType SensorType { get; set; }
+        public byte I2cDataLength { get; set; }
+    }
+
+    /// <summary>
+    /// The type of Groove sensors
+    /// </summary>
+    public enum GrooveSensorType
```

Comment:
normally this should be type per file (VS should auto fix it correctly with Ctrl-. and then setting the right item from the list)